wallet-new/src/Cardano/Wallet/Kernel/Wallets.hs

```diff
@@ -138,11 +150,8 @@ createHdWallet pw mnemonic spendingPassword assuranceLevel walletName = do
                              -- See preconditon above.
                              (\hdRoot -> Left $ CreateHdWallet hdRoot mempty)
     case res of
-         Left e   -> return . Left $ CreateWalletFailed e
-         Right hdRoot -> do
-             -- STEP 3: Insert the 'EncryptedSecretKey' into the 'Keystore'
-             Keystore.insert (WalletIdHdRnd newRootId) esk (pw ^. walletKeystore)
-             return (Right hdRoot)
+         Left e       -> return . Left $ CreateWalletFailed e
```

Comment:
TODO: Let's turn the generic `e` into the (sole) type constructor for `HD.CreateHdRootError` and add an explicit comment regarding atomicity considerations.